tempesta_fw/http.c

```diff
@@ -2998,6 +3006,11 @@ tfw_http_popreq(TfwHttpMsg *hmresp)
 		srv_conn->msg_sent = NULL;
 	__http_req_delist(srv_conn, req);
 	tfw_http_conn_nip_adjust(srv_conn);
+
+	if (unlikely(!fwd_unsent)) {
+		spin_unlock(&srv_conn->fwd_qlock);
+		return;
+	}
```

Comment:
`fwd_unsent` is used to execute the small code above only and exit the function. Moreover, the code above is `http_req_delist()` with zeroing and `tfw_http_conn_nip_adjust()` call. I propose to leave `http_req_delist()` as
```C
http_req_delist()
{
    if ((TfwMsg *)req == srv_conn->msg_sent)
        srv_conn->msg_sent = NULL;
    __http_req_delist(srv_conn, req);
    tfw_http_conn_nip_adjust(srv_conn);
```
call it here and define
```C
http_req_delist_locked()
{
    spin_lock(&srv_conn->fwd_qlock);
    http_req_delist();
    spin_unlock(&srv_conn->fwd_qlock);
}
```
and call it instead of `tfw_http_popreq( .., false)`